bump.lua

```diff
@@ -481,15 +515,23 @@ function World:project(item, x,y,w,h, goalX, goalY, filter)
           col.type     = responseName
 
           len = len + 1
+          if collisions == nil then
+            collisions = {}
+          end
           collisions[len] = col
         end
       end
     end
   end
 
-  table.sort(collisions, sortByTiAndDistance)
+  Pool.free(visited)
+  Pool.free(dictItemsInCellRect)
 
-  return collisions, len
+  if collisions ~= nil then
+    table.sort(collisions, sortByTiAndDistance)
+  end
+
```

Comment:
EMPTY_TABLE is by far the newest change here, everything else I changed back in march. The idea is that we reuse one static `{}` whenever we return results with no collisions. Formerly, for my own use, I was just returning `nil, 0`. But I think this is a little nicer because there's no API change.

It's ever so slightly dangerous though, because if the user does something crazy and inserts into EMPTY_TABLE, that could cause some incredibly strange behavior.